google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/Type.java

```diff
@@ -87,13 +92,48 @@ public static Aggregate int64Sum() {
   }
 
   /** Creates an Aggregate type with a SUM aggregator and specified input type. */
-  public static Aggregate sum(SumAggregateInput inputType) {
+  public static <Input extends Type & SumAggregateInput> Aggregate sum(Input inputType) {
     return Aggregate.create(inputType, Aggregate.Aggregator.Sum.create());
   }
 
+  @Deprecated
+  public static Aggregate sum(SumAggregateInput inputType) {
+    return Aggregate.create((Type) inputType, Aggregate.Aggregator.Sum.create());
+  }
+
+  /** Creates an Aggregate type with a MIN aggregator and Int64 input type. */
+  public static Aggregate int64Min() {
+    return min(bigEndianInt64());
+  }
+
+  /** Creates an Aggregate type with a MIN aggregator and specified input type. */
+  public static <Input extends Type & MinAggregateInput> Aggregate min(Input inputType) {
+    return Aggregate.create(inputType, Aggregate.Aggregator.Min.create());
+  }
+
+  /** Creates an Aggregate type with a MAX aggregator and Int64 input type. */
+  public static Aggregate int64Max() {
+    return max(bigEndianInt64());
+  }
+
+  /** Creates an Aggregate type with a MAX aggregator and specified input type. */
+  public static <Input extends Type & MaxAggregateInput> Aggregate max(Input inputType) {
+    return Aggregate.create(inputType, Aggregate.Aggregator.Max.create());
+  }
+
+  /** Creates an Aggregate type with a HLL aggregator and Int64 input type. */
+  public static Aggregate rawBytesHll() {
+    return hll(rawBytes());
+  }
+
+  /** Creates an Aggregate type with a HLL aggregator and specified input type. */
+  public static <Input extends Type & HllAggregateInput> Aggregate hll(Input inputType) {
+    return Aggregate.create(inputType, Aggregate.Aggregator.Hll.create());
+  }
+
   /** Represents a string of bytes with a specific encoding. */
   @AutoValue
-  public abstract static class Bytes extends Type {
+  public abstract static class Bytes extends Type implements HllAggregateInput {
```

Comment:
same, we only support int64 inputs right now so we shouldn't have this here.

Reply:
fixed